backend/src/nodes/utils/auto_split.py

```diff
@@ -12,6 +12,34 @@ class Split:
     pass
 
 
+def estimate_tile_size(
+    free: int,
+    total: int,
+    model_size: int,
+    img: np.ndarray,
+    img_element_size: int = 4,
+    allowed_usage: float = 0.8,
+) -> Union[int, None]:
+    h, w, c = get_h_w_c(img)
+    img_bytes = h * w * c * img_element_size
+    mem_required_estimation = (model_size / (1024 * 52)) * img_bytes
+
+    tile_pixels = w * h * (free * allowed_usage) / mem_required_estimation
+    # the largest power-of-2 tile_size such that tile_size**2 < tile_pixels
+    tile_size = 2 ** (int(tile_pixels**0.5).bit_length() - 1)
```

Comment:
We talked about this on discord: I also rounded it down to a power of 2 is because I wanted to avoid OOM. I judged that an OOM is more expensive than a smaller tile size.

If you want to remove the power of 2, make it `tile_size = int(tile_pixels**0.5)`.